### PR TITLE
[feature] Added static files minification and cache invalidation #216

### LINKED ISSUE #216: [feature] Implement static minification and cache invalidation
We need implement static asset minification and make sure Javascript and CSS files get updated automatically without having to ask users to refresh their browser cache.

Unfortunately django-pipeline does not support simple minification without bundling files together, which requires changes to the code of OpenWISP modules which are overkill (given also HTTP2 multiplexing it's not extremely needed).

But its feature to Minify HTML is indeed useful, so we can use a mixture of different tools.

Packages to install: `pip install django-pipeline django-compress-staticfiles`

Configuration example:

```python
# HTML minification with django pipeline
INSTALLED_APPS.append('pipeline')
MIDDLEWARE.append('pipeline.middleware.MinifyHTMLMiddleware')
PIPELINE = {
    'PIPELINE_ENABLED': True
}

# static files minification and invalidation with django-compress-staticfiles
STATICFILES_STORAGE = 'compress_staticfiles.storage.CompressStaticFilesStorage'
# GZIP compression is handled by nginx
BROTLI_STATIC_COMPRESSION = False
GZIP_STATIC_COMPRESSION = False
```

When using this configuration, a bug in openwisp modules is surfacing, we should not use the `static()` function in admin class media definitions, hence this has to be fixed/checked in the following modules:

- [x] django-x509 (https://github.com/openwisp/django-x509/commit/264cfb8372bebb12859c79834584c2ed0265230f)
- [x] django-loci
- [x] openwisp-utils
- [x] openwisp-users
- [x] openwisp-controller (https://github.com/openwisp/openwisp-controller/commit/235ec6f34e9af3e46b3812ea717f6b66fea21687)
- [x] openwisp-network-topology (https://github.com/openwisp/openwisp-network-topology/pull/96)
- [x] openwisp-radius (https://github.com/openwisp/openwisp-radius/pull/187)
- [x] openwisp-monitoring
- [x] openwisp-ipam
- [x] openwisp-firmware-upgrader (https://github.com/openwisp/openwisp-firmware-upgrader/commit/be3a6bb96520b0bf4462028de56c4e716b1fcf4c)

### PR DESCRIPTION
**Pull request by pandafy**
Closes #216

-------------

**Blockers**
 - Requires new releases of 
     - [x] openwisp-controller
     - [x] openwisp-network-topology
 - [x] https://github.com/openwisp/openwisp-network-topology/pull/100

----------

I have used the following playbook to test the role on Ubuntu 20.04 

```yaml
- hosts: openwisp2
  become: "{{ become | default('yes') }}"
  roles:
    - openwisp.openwisp2
  vars:
    openwisp2_default_from_email: "openwisp2@openwisp2.mydomain.com"

    openwisp2_controller_pip: https://github.com/openwisp/openwisp-controller/tarball/master
    openwisp2_users_pip: https://github.com/openwisp/openwisp-users/tarball/master
    openwisp2_utils_pip: https://github.com/openwisp/openwisp-utils/tarball/master
    openwisp2_django_netjsonconfig_pip: https://github.com/openwisp/django-netjsonconfig/tarball/master
    openwisp2_django_x509_pip: https://github.com/openwisp/django-x509/tarball/master
    openwisp2_netjsonconfig_pip: https://github.com/openwisp/netjsonconfig/tarball/master
    openwisp2_network_topology_pip: https://github.com/openwisp/openwisp-network-topology/tarball/master

    openwisp2_extra_python_packages:
      - "{{ openwisp2_controller_pip }}"
      - "{{ openwisp2_users_pip }}"
      - "{{ openwisp2_utils_pip }}"
      - "{{ openwisp2_django_netjsonconfig_pip }}"
      - "{{ openwisp2_django_x509_pip }}"
      - "{{ openwisp2_netjsonconfig_pip }}"
      - "{{ openwisp2_network_topology_pip }}"
```

### CODE REVIEW

**Comment by pandafy** [comment]
Build is failing due to Idempotence tests.

File: templates/openwisp2/settings.py [hunk 1]
```diff
@@ -293,6 +293,19 @@
     }
 }
 
+# HTML minification with django pipeline
+INSTALLED_APPS.append('pipeline')
+MIDDLEWARE.append('pipeline.middleware.MinifyHTMLMiddleware')
```
Comment (atb00ker) [inline templates/openwisp2/settings.py:298]: Why not add them in `INSTALLED_APPS` and `MIDDLEWARE` list blocks above?

**Review by atb00ker — COMMENTED** [review]
Great Going! :+1:

File: templates/openwisp2/settings.py (hunk 1 above)
Comment (pandafy) [reply templates/openwisp2/settings.py:298]: Thought adding them in a separate block will make more sense. I will update it.

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**